src/pv/pvSupport.h

```diff
@@ -38,17 +38,17 @@ class epicsShareClass PVSupport
      */
     virtual ~PVSupport(){}
     /**
-     * @brief Optional  initialization method.
+     * @brief Required initialization method.
      *
-     * Called after PVRecord is created but before record is installed into PVDatabase.
+     * Implementation classes must define an init() method that must be
+     * explicitly called by PVRecord classes that use them to provide
+     * references to the specific fields to be supported. Different support
+     * will may different fields, so a virtual method cannot be defined in
+     * this base class to support them, hence this method always fails.
      *
-     * @param pvValue The field to support.
-     * @param pvSupport Support specific fields.
      * @return <b>true</b> for success and <b>false</b> for failure.
      */
-    virtual bool init(
-        epics::pvData::PVFieldPtr const & pvValue,
-        epics::pvData::PVFieldPtr const & pvSupport) {return true;}
+    bool init() {return false;}
```

Comment:
I am unsure what the original intent was here, but if all classes that inherit from PVSupport were supposed to have init(pvValue, pvSupport), I do not see a problem with leaving this (and ControlSupport class) as it was, and only changing/renaming init() method on line 48 in ScalarAlarmSupport, which would presumably fix the warnings, and would be a smaller API change. I do not quite see a point of introducing a new init() method here that is not being used. Sorry if I am missing something...

Reply:
I think Marty was suggesting that classes derived from PVSupport would provide their own `init()` method to set up the their added member data, as his `class ScalarAlarmSupport` (in `src/pv/scalarAlarmSupport.{h,cpp}`) does. His Doxygen comment said the method would be optional (line 41 above) implying that some might not need to, but I think that in most cases the derived class would need additional parameters anyway (as `class ScalarAlarmSupport` does), so it doesn't actually make sense for `init()` to be a virtual method with a fixed set of parameters anyway.

My change was written to try and keep what I thought his design intent was. If we don't care about that then I agree that these changes to the base class are unnecessary.